crates/ironrdp-web/src/clipboard/mod.rs

```diff
@@ -0,0 +1,565 @@
+    pub name: &'static str,
+}
+
+impl ClientFormatDescriptor {
+    pub const fn new(id: ClipboardFormatId, name: &'static str) -> Self {
+        Self { id, name }
+    }
+}
+
+impl From<ClientFormatDescriptor> for ClipboardFormat {
+    fn from(descriptor: ClientFormatDescriptor) -> Self {
+        ClipboardFormat::new(descriptor.id).with_name(ClipboardFormatName::new_static(descriptor.name))
+    }
+}
+
+const FORMAT_WIN_HTML_ID: ClipboardFormatId = ClipboardFormatId(0xC001);
+const FORMAT_MIME_HTML_ID: ClipboardFormatId = ClipboardFormatId(0xC002);
+const FORMAT_PNG_ID: ClipboardFormatId = ClipboardFormatId(0xC003);
+const FORMAT_MIME_PNG_ID: ClipboardFormatId = ClipboardFormatId(0xC004);
+
+const FORMAT_WIN_HTML_NAME: &str = "HTML Format";
+const FORMAT_MIME_HTML_NAME: &str = "text/html";
+const FORMAT_PNG_NAME: &str = "PNG";
+const FORMAT_MIME_PNG_NAME: &str = "image/png";
+
+const FORMAT_WIN_HTML: ClientFormatDescriptor = ClientFormatDescriptor::new(FORMAT_WIN_HTML_ID, FORMAT_WIN_HTML_NAME);
+const FORMAT_MIME_HTML: ClientFormatDescriptor =
+    ClientFormatDescriptor::new(FORMAT_MIME_HTML_ID, FORMAT_MIME_HTML_NAME);
+const FORMAT_PNG: ClientFormatDescriptor = ClientFormatDescriptor::new(FORMAT_PNG_ID, FORMAT_PNG_NAME);
+const FORMAT_MIME_PNG: ClientFormatDescriptor = ClientFormatDescriptor::new(FORMAT_MIME_PNG_ID, FORMAT_MIME_PNG_NAME);
+
+/// Message proxy used to send clipboard-related messages to the application main event loop
+#[derive(Debug, Clone)]
+pub struct WasmClipboardMessageProxy {
+    tx: mpsc::UnboundedSender<RdpInputEvent>,
+}
+
+impl WasmClipboardMessageProxy {
+    pub fn new(tx: mpsc::UnboundedSender<RdpInputEvent>) -> Self {
+        Self { tx }
+    }
+
+    /// Send messages which require action on CLIPRDR SVC
+    pub fn send_cliprdr_message(&self, message: ClipboardMessage) {
+        if self.tx.unbounded_send(RdpInputEvent::Cliprdr(message)).is_err() {
+            error!("Failed to send os clipboard message, receiver is closed");
+        }
+    }
+
+    /// Send messages which require action on wasm clipboard backend
+    pub fn send_backend_message(&self, message: WasmClipboardBackendMessage) {
+        if self
+            .tx
+            .unbounded_send(RdpInputEvent::ClipboardBackend(message))
+            .is_err()
+        {
+            error!("Failed to send os clipboard message, receiver is closed");
+        }
+    }
+}
+
+/// Messages sent by the JS code or CLIPRDR to the backend implementation.
+#[derive(Debug)]
+pub enum WasmClipboardBackendMessage {
+    LocalClipboardChanged(ClipboardTransaction),
+    RemoteDataRequest(ClipboardFormatId),
+
+    RemoteClipboardChanged(Vec<ClipboardFormat>),
+    RemoteDataResponse(FormatDataResponse<'static>),
+
+    FormatListReceived,
+    ForceClipboardUpdate,
+}
+
+/// Clipboard backend implementation for web. This object should be created once per session and
+/// kept alive until session is terminated.
+pub struct WasmCipboard {
```

Comment:
```suggestion
pub struct WasmClipboard {
```

Reply:
Huh... Interesting why `typos` did not catch that even though I ran it multiple times... Maybe I need to update it...

Reply:
Oh, I thought it would indeed!
Looks like they don’t have this typo in their database: https://github.com/epage/typos/blob/194cd85dd6eecb097568b803d374ab21dccace1b/crates/typos-dict/assets/words.csv#L10034
I commented their issue for the November batch: https://github.com/crate-ci/typos/issues/865#issuecomment-1792626518
Should be properly detected next time :slightly_smiling_face: